targets/targets.json

```diff
@@ -5318,7 +5318,8 @@
         "inherits": ["MCU_NRF51_16K_S110"],
         "macros_add": ["TARGET_NRF_LFCLK_RC"],
         "release_versions": ["2"],
-        "device_name": "nRF51822_xxAA"
+        "device_name": "nRF51822_xxAA",
+        "supported_toolchains": ["ARMC5", "GCC_ARM"]
```

Comment:
Why only ARMC and GCC and not IAR?

Reply:
`NRF51_MICROBIT` inherits from `MCU_NRF51_16K_S110` which inherits from `MCU_NRF51_S110` and `MCU_NRF51_16K_BASE`. `MCU_NRF51_S110` is a dead end and does not define a toolchain. So following `MCU_NRF51_16K_BASE` this inherits from `MCU_NRF51`, which defines the `supported_toolchains` as `[“ARM", "GCC_ARM”]`. This means before my change the supported toolchains were Arm C6 and GCC ARM, but before the change which broke microbit they should have been Arm C5 and GCC ARM. This definition seems to be setup in 2016, which means as best as I can tell this platform has never declared support for IAR.

Reply:
I had to check this one as well as I would have assumed nrf51 was supported also by IAR but has not been.